### PR TITLE
CAN - 111

### PR DESCRIPTION
**Pull request by ashish76**
1. CAN - 326 Create API for social list
2. CAN - 327 Create link and unlink API

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -1078,43 +1079,71 @@ public function socialCallback(Request $request, Validate $validate)
...
-                $status = 200;
-                $message = trans('message.success.success');
+            $social_user = SocialUser::where(['social_email' => $user_email, 'provider' => $provider])->first();
+            if($request->user()){
+                if (isset($social_user) && isset($social_user->user_id)) {
+                    $status = 403;
+                    $message = trans('message.social.already_linked');
+                    $data = null;
+				}else{
+                    $socialUser = SocialUser::create([
```
Comment (akuks) [inline app/Http/Controllers/UserController.php:1089]: This is not seems to be used any where

Comment (akuks) [inline app/Http/Controllers/UserController.php:1089]: Social::Create command is repeated

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -1078,43 +1079,71 @@ public function socialCallback(Request $request, Validate $validate)
...
-            $generateToken = Util::httpPost($postUrl, $payload);
-            if($generateToken->status_code == 200){
-                $data = [
-                    "auth" => $generateToken->data,
-                    "user" => new UserResource($user),
-                ];
-                $status = 200;
-                $message = trans('message.success.success');
+            $social_user = SocialUser::where(['social_email' => $user_email, 'provider' => $provider])->first();
+            if($request->user()){
```
Comment (akuks) [inline app/Http/Controllers/UserController.php:1083]: Can you please move multiple if else in a separate function or may be further break it down. As it is not very readable.

**Review by akuks — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -1078,43 +1080,41 @@ public function socialCallback(Request $request, Validate $validate)
...
+                    'username' => $user->email,
+                    'password' => env('PASSPORT_MASTER_PASSWORD'),
+                    'scope' => '*',
                 ];
-                $status = 200;
-                $message = trans('message.success.success');
-                return $this->resProvider->apiJsonResponse($status, $message, $data, null);
-            }
+                $generateToken = Util::httpPost($postUrl, $payload);
+                if($generateToken->status_code == 200){
```
Comment (akuks) [inline app/Http/Controllers/UserController.php:1109]: Code from 1109-1119 is same as 464-474

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -1291,4 +1291,443 @@ public function reSendOtp(Request $request, Validate $validate)
...
+     *     description="Exception Throwable",
+     *     @OA\JsonContent(
+     *          oneOf={@OA\Schema(ref="#/components/schemas/ExceptionRes")}
+     *     )
+     *   )
+     *
+     * )
+     */
+
+    public function SocialLink(Request $request, Validate $validate)
```
Comment (akuks) [inline app/Http/Controllers/UserController.php:1560]: Functioname should be in camelcase

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -1291,4 +1291,443 @@ public function reSendOtp(Request $request, Validate $validate)
...
+     *     response=403,
+     *     description="Exception Throwable",
+     *     @OA\JsonContent(
+     *          oneOf={@OA\Schema(ref="#/components/schemas/ExceptionRes")}
+     *     )
+     *   )
+     * )
+     */
+
+    public function SocialDelete(Request $request, $id)
```
Comment (akuks) [inline app/Http/Controllers/UserController.php:1465]: Functioname should be in camelcase

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -1291,4 +1291,443 @@ public function reSendOtp(Request $request, Validate $validate)
...
+                return $this->resProvider->apiJsonResponse($status, $message, $data, null);
+
+        }catch (Exception $ex) {
+            $status = 400;
+            $message = trans('message.error.exception');
+            return $this->resProvider->apiJsonResponse($status, $message, null, null);
+        }
+    }
+
+    protected function CreateSocialUser($data,$provider,$userId)
```
Comment (akuks) [inline app/Http/Controllers/UserController.php:1600]: Functioname should be in camelcase

**Review by akuks — CHANGES_REQUESTED** [review]
Please go through the changes

File: app/Http/Controllers/UserController.php (hunk 6 above)
Comment (ashish76) [reply app/Http/Controllers/UserController.php]: Done

File: app/Http/Controllers/UserController.php (hunk 5 above)
Comment (ashish76) [reply app/Http/Controllers/UserController.php]: Done

File: app/Http/Controllers/UserController.php (hunk 4 above)
Comment (ashish76) [reply app/Http/Controllers/UserController.php]: Done

File: app/Http/Controllers/UserController.php (hunk 3 above)
Comment (ashish76) [reply app/Http/Controllers/UserController.php]: Done

**Pull request merged**